openmc/data/photonuclear.py

```diff
@@ -0,0 +1,879 @@
+
+    def __iter__(self):
+        return iter(self.reactions.values())
+
+    @property
+    def atomic_number(self):
+        return self._atomic_number
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def mass_number(self):
+        return self._mass_number
+
+    @property
+    def metastable(self):
+        return self._metastable
+
+    @property
+    def atomic_weight_ratio(self):
+        return self._atomic_weight_ratio
+
+    @property
+    def atomic_symbol(self):
+        return ATOMIC_SYMBOL[self.atomic_number]
+
+    @name.setter
+    def name(self, name):
+        cv.check_type("name", name, str)
+        self._name = name
+
+    @atomic_number.setter
+    def atomic_number(self, atomic_number):
+        cv.check_type("atomic number", atomic_number, Integral)
+        cv.check_greater_than("atomic number", atomic_number, 0, True)
+        self._atomic_number = atomic_number
+
+    @mass_number.setter
+    def mass_number(self, mass_number):
+        cv.check_type("mass number", mass_number, Integral)
+        cv.check_greater_than("mass number", mass_number, 0, True)
+        self._mass_number = mass_number
+
+    @metastable.setter
+    def metastable(self, metastable):
+        cv.check_type("metastable", metastable, Integral)
+        cv.check_greater_than("metastable", metastable, 0, True)
+        self._metastable = metastable
+
+    @atomic_weight_ratio.setter
+    def atomic_weight_ratio(self, atomic_weight_ratio):
+        cv.check_type("atomic weight ratio", atomic_weight_ratio, Real)
+        cv.check_greater_than("atomic weight ratio", atomic_weight_ratio, 0.0)
+        self._atomic_weight_ratio = atomic_weight_ratio
+        
+    @property
+    def fission_energy(self):
+        return self._fission_energy
+
+    @fission_energy.setter
+    def fission_energy(self, fission_energy):
+        cv.check_type('fission energy release', fission_energy,
+                      FissionEnergyRelease)
+        self._fission_energy = fission_energy        
+
+    @classmethod
+    def from_endf(cls, ev_or_filename):
+        """Generate incident photo-nuclear data from an ENDF evaluation
+
+        Parameters
+        ----------
+        ev_or_filename : openmc.data.endf.Evaluation or str
+            ENDF evaluation to read from. If given as a string, it is assumed to
+            be the filename for the ENDF file.
+
+
+        Returns
+        -------
+        openmc.data.Photonuclear
+            photo-nuclear interaction data
+
+        """
+
+        if isinstance(ev_or_filename, Evaluation):
+            ev = ev_or_filename
+        else:
+            ev = Evaluation(ev_or_filename)
+
+        atomic_number = ev.target["atomic_number"]
+        mass_number = ev.target["mass_number"]
+        metastable = ev.target["isomeric_state"]
+        atomic_weight_ratio = ev.target["mass"]
+
+        element = ATOMIC_SYMBOL[atomic_number]
+        name = "{}{}".format(element, mass_number)
```

Comment:
The standard naming convention (e.g., Am242_m1) requires including the metastable level. The original code would name Am-242m and Am-242 the same, causing conflicts.
```suggestion
        name = "{}{}".format(element, mass_number)
        if metastable > 0:
            name += f'_m{metastable}'
```

Reply:
Nice find! I decided to copy the implementation in IncidentNeutron.